src/link.cpp

```diff
@@ -1122,53 +1122,127 @@ static bool zig_lld_link(ZigLLVM_ObjectFormatType oformat, const char **args, si
 }
 
 static void add_uefi_link_args(LinkJob *lj) {
-    lj->args.append("/BASE:0");
-    lj->args.append("/ENTRY:EfiMain");
-    lj->args.append("/OPT:REF");
-    lj->args.append("/SAFESEH:NO");
-    lj->args.append("/MERGE:.rdata=.data");
-    lj->args.append("/ALIGN:32");
-    lj->args.append("/NODEFAULTLIB");
-    lj->args.append("/SECTION:.xdata,D");
+    lj->args.append("-BASE:0");
+    lj->args.append("-ENTRY:EfiMain");
+    lj->args.append("-OPT:REF");
+    lj->args.append("-SAFESEH:NO");
+    lj->args.append("-MERGE:.rdata=.data");
+    lj->args.append("-ALIGN:32");
+    lj->args.append("-NODEFAULTLIB");
+    lj->args.append("-SECTION:.xdata,D");
 }
 
-static void add_nt_link_args(LinkJob *lj, bool is_library) {
+static void add_msvc_link_args(LinkJob *lj, bool is_library) {
     CodeGen *g = lj->codegen;
 
-    if (lj->link_in_crt) {
-        // TODO: https://github.com/ziglang/zig/issues/2064
-        bool is_dynamic = true; // g->is_dynamic;
-        const char *lib_str = is_dynamic ? "" : "lib";
-        const char *d_str = (g->build_mode == BuildModeDebug) ? "d" : "";
-
-        if (!is_dynamic) {
-            Buf *cmt_lib_name = buf_sprintf("libcmt%s.lib", d_str);
-            lj->args.append(buf_ptr(cmt_lib_name));
+    // TODO: https://github.com/ziglang/zig/issues/2064
+    bool is_dynamic = true; // g->is_dynamic;
+    const char *lib_str = is_dynamic ? "" : "lib";
+    const char *d_str = (g->build_mode == BuildModeDebug) ? "d" : "";
+
+    if (!is_dynamic) {
+        Buf *cmt_lib_name = buf_sprintf("libcmt%s.lib", d_str);
+        lj->args.append(buf_ptr(cmt_lib_name));
+    } else {
+        Buf *msvcrt_lib_name = buf_sprintf("msvcrt%s.lib", d_str);
+        lj->args.append(buf_ptr(msvcrt_lib_name));
+    }
+
+    Buf *vcruntime_lib_name = buf_sprintf("%svcruntime%s.lib", lib_str, d_str);
+    lj->args.append(buf_ptr(vcruntime_lib_name));
+
+    Buf *crt_lib_name = buf_sprintf("%sucrt%s.lib", lib_str, d_str);
+    lj->args.append(buf_ptr(crt_lib_name));
+
+    //Visual C++ 2015 Conformance Changes
+    //https://msdn.microsoft.com/en-us/library/bb531344.aspx
+    lj->args.append("legacy_stdio_definitions.lib");
+
+    // msvcrt depends on kernel32 and ntdll
+    lj->args.append("kernel32.lib");
+    lj->args.append("ntdll.lib");
+}
+
+static const char *get_libc_file(ZigLibCInstallation *lib, const char *file) {
+    Buf *out_buf = buf_alloc();
+    os_path_join(&lib->crt_dir, buf_create_from_str(file), out_buf);
+    return buf_ptr(out_buf);
+}
+
+static const char *get_libc_static_file(ZigLibCInstallation *lib, const char *file) {
+    Buf *static_crt_dir = buf_alloc();
+    Buf *out_buf = buf_alloc();
+    if (zig_libc_cc_print_file_name(file, static_crt_dir, true, true) != ErrorNone) {
```

Comment:
The call to `zig_libc_cc_print_file_name` needs to be called during the libc detection in libc_installation.cpp. Once we get to link.cpp we should only be appending file names to known dir paths.

Reply:
this is here because zig no longer stores the path for the static files, because they are built from source now on linux. i wasn't sure if i should introduce an entire entry to the libc file, just to accommodate mingw.

Reply:
I do think that's the right way to do it. It's ok to have a field of libc_installation that is only for mingw.